tools/bin/mbedtls-rewrite-branch-style

```diff
@@ -0,0 +1,586 @@
+        """Rewrite branch on top of the code style switch commit.
+
+        1. Rebase the head of branch onto the last commit with the old style.
+        2. Rewrite all new commits on top of the code style switch commit.
+
+        This method assumes it has ownership of the current worktree.
+        It does not create or modify any branch name.
+        """
+        switch_commit = self.find_switch_commit()
+        last_old_commit = switch_commit + '~1'
+        # Phase 1: rebase onto last_old_commit
+        self.info('Rebasing {} onto {}', self.head_sha(), last_old_commit)
+        self.run_git(['rebase', last_old_commit])
+        self.info('Head after rebase: {}', self.head_sha())
+        # Phase 2: Iterate over the new commits
+        new_commits = self.list_commit_range(last_old_commit, 'HEAD')
+        self.info('Will restyle {} new commits ({})',
+                  len(new_commits), ' '.join(new_commits))
+        self.info('Starting from code style switch commit {}', switch_commit)
+        self.run_git_checkout(switch_commit)
+        if self.working_branch_name:
+            self.run_git(['branch', '--force', self.working_branch_name, 'HEAD'])
+            self.run_git(['checkout', '--quiet', self.working_branch_name])
+        for new_commit in new_commits:
+            self.restyle_commit_onto_current(new_commit)
+
+def main() -> int:
+    """Command line entry point."""
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument('--backup-branch', metavar='BRANCH_NAME',
+                        default='old-code-style/{}',
+                        help='Back up the current history'
+                             ' (default: old-code-style/OLD_NAME)'
+                             ' (empty: do not back up)')
+    parser.add_argument('--onto', '-t', metavar='BRANCH',
+                        help='Branch to rewrite onto'
+                             ' (default/empty: guess development or mbedtls-2.28)')
+    parser.add_argument('--new-branch-name', '-c', metavar='BRANCH_NAME',
+                        default='',
+                        help='Use this branch for the new work (retained on failure),'
+                             ' do not change the original branch'
+                             ' (default/empty: update original branch on success)')
+    parser.add_argument('--verbose', '-v',
+                        action='store_true',
+                        help='be more verbose')
+    parser.add_argument('branch',
+                        nargs='?', default='',
+                        help='Branch to rewrite with code style changes'
+                             ' (default: branch checked out in the current directory)')
+    args = parser.parse_args()
+    branch_rewriter = CodeStyleBranchRewriter(
+        working_branch_name=args.new_branch_name,
+        verbose=args.verbose)
+    if not branch_rewriter.system_is_ok():
+        return 2
+    if not args.branch:
```

Comment:
After checking that system has the required tools, we could also check that the branch is actually an old code style branch.

`git log --grep="Switch to the new code style" --format="%H"` will return the hash of the style transition commit on the current brach or empty if in the old style.